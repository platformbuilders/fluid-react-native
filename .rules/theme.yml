rules:
  - id: avoid-scale-functions-react-native
    languages: [javascript, typescript]
    message: 'Evite usar funções como scale ou moderateScale para definir width ou height. O React Native já faz o scaling nativamente.'
    severity: WARNING
    patterns:
      - pattern-regex: "(moderateScale|scale)\\(\\d+\\)"
    paths:
      exclude:
        - '*.json'
    metadata:
      description: 'Esta regra sugere evitar o uso de funções como scale ou moderateScale, já que o React Native já faz o scaling de forma nativa, tornando o código mais simples e performático.'
      tags: ['width', 'height', 'responsividade', 'react-native']

  - id: avoid-stylesheet-use-styled-components
    languages: [javascript, typescript]
    message: 'Evite usar StyleSheet.create. Utilize styled-components para definir estilos.'
    severity: WARNING
    patterns:
      - pattern-regex: "StyleSheet\\.create"
    paths:
      exclude:
        - '*.json'
        - 'src/components/PinInput/lib/react-native-smooth-pincode-input/SmoothPinCodeInput.js'
    metadata:
      description: 'Esta regra proíbe o uso de StyleSheet.create no React Native e recomenda a utilização de styled-components para definir estilos, promovendo uma abordagem mais moderna e flexível.'
      tags: ['styles', 'react-native', 'styled-components']

  - id: avoid-direct-theme-import
    languages: [javascript, typescript]
    message: 'Importação direta do tema não é permitida. Utilize um módulo intermediário para gerenciar temas.'
    severity: WARNING
    patterns:
      - pattern-regex: "import\\s?{.*}\\s?from\\s?'~\/theme'"
    paths:
      exclude:
        - '*.json'
    metadata:
      description: "Esta regra impede a importação direta do tema da aplicação ('~/theme'). É recomendada a utilização de módulos intermediários para gerenciar e manter a modularidade do tema no código."
      tags: ['theme', 'import', 'modularidade']

  - id: prefer-hairlinewidth-over-1px
    languages: [javascript, typescript]
    message: 'O ideal seria usar StyleSheet.hairlineWidth em vez de utilizar 1px ou 1 diretamente. Isso garante consistência de design em múltiplas resoluções.'
    severity: WARNING
    patterns:
      - pattern-regex: "height:\\s?(\\${moderateScale\\(1\\)}|\\${scale\\(1\\)\\}|1;|1\\s|1px)"
    paths:
      exclude:
        - '*.json'
    metadata:
      description: 'Esta regra sugere o uso de StyleSheet.hairlineWidth ao invés de definir height como 1 ou 1px diretamente, garantindo melhor consistência e adaptabilidade do design entre diferentes resoluções de tela.'
      tags: ['styles', 'height', 'hairlineWidth']

  - id: avoid-scale-or-moderatescale-in-fontsize
    languages: [javascript, typescript]
    message: 'Evite usar moderateScale ou scale no fontSize. O React Native já faz o escalonamento de fontes de forma nativa.'
    severity: WARNING
    patterns:
      - pattern-regex: "(fontSize:\\s?\\${moderateScale\\(\\d+\\)}|fontSize:\\s?\\${scale\\(\\d+\\)})"
    paths:
      exclude:
        - '*.json'
    metadata:
      description: 'Essa regra proíbe o uso de moderateScale ou scale para definir o tamanho de fonte. O React Native já realiza o escalonamento de fontes de forma nativa, garantindo consistência entre dispositivos.'
      tags: ['styles', 'fontsize', 'scale', 'moderatescale']

  - id: avoid-styled-text-use-typography
    languages: [javascript, typescript]
    message: 'Evite usar styled.Text. Use o componente Typography, que contém lógica de renderização e sizing de fontes.'
    severity: WARNING
    patterns:
      - pattern-regex: "styled\\.Text`"
    paths:
      exclude:
        - '*.json'
    metadata:
      description: 'O componente Typography deve ser utilizado no lugar de styled.Text, pois ele já contém a lógica de renderização e sizing de fontes, garantindo consistência e eficiência no código.'
      tags: ['styled-components', 'typography', 'fonts']

  - id: prefer-font-weight-bold
    languages: [javascript, typescript]
    message: 'Recomenda-se usar font-weight: bold; para evitar problemas de renderização no Android.'
    severity: WARNING
    patterns:
      - pattern-regex: "font-weight:\\s?600"
    paths:
      exclude:
        - '*.json'
    metadata:
      description: 'Evite usar font-weight: 600 em estilos, pois pode causar problemas de renderização em dispositivos Android. Use font-weight: bold para garantir uma exibição consistente.'
      tags: ['styles', 'font-weight', 'android', 'bold']

  - id: prefer-correct-border-radius-from-theme
    languages: [javascript, typescript]
    message: "Recomenda-se usar valores de border-radius obtidos corretamente do tema, por meio de getTheme('borderRadius')."
    severity: WARNING
    patterns:
      - pattern-regex: "const\\s+\\w+\\s?=\\s?getTheme\\('border\\w+'\\);"
    paths:
      exclude:
        - '*.json'
    metadata:
      description: "Garanta que os valores de border-radius sejam obtidos do tema corretamente utilizando getTheme('borderRadius'). Isso assegura consistência no design."
      tags: ['styles', 'border-radius', 'theme']

  - id: prefer-theme-colors
    languages: [javascript, typescript]
    message: 'Sempre utilize cores definidas no tema. Evite o uso de códigos hexadecimais hardcoded.'
    severity: WARNING
    patterns:
      - pattern-regex: ".*?:\\s?#([A-Fa-f0-9]{6}|[A-Fa-f0-9]{3})"
    paths:
      exclude:
        - '*.json'
    metadata:
      description: 'As cores devem ser obtidas a partir do tema da aplicação, garantindo consistência e facilitando a manutenção. Evite usar códigos hexadecimais diretamente.'
      tags: ['styles', 'colors', 'theme']

  - id: use-theme-spacing
    languages: [javascript, typescript]
    message: "Utilize valores de spacing vindos do tema corretamente, como 'padding: ${smallSpacing}px'."
    severity: WARNING
    patterns:
      - pattern-regex: "(padding|margin)(-?)(left|right|top|bottom|vertical|horizontal)?:\\s?(\\d+px);"
    paths:
      exclude:
        - '*.json'
    metadata:
      description: 'Os valores de padding e margin devem ser obtidos diretamente do tema da aplicação, garantindo consistência no espaçamento entre os elementos. Evite usar valores fixos em pixels.'
      tags: ['styles', 'spacing', 'theme', 'padding', 'margin']

  - id: use-variant-instead-of-fontsize
    languages: [javascript, typescript]
    message: "Utilize a prop 'variant' do componente Typography em vez de definir 'font-size' manualmente."
    severity: WARNING
    patterns:
      - pattern-regex: 'font-size'
    paths:
      exclude:
        - '*.json'
        - 'src/components/Avatar/styles.ts'
        - 'src/components/Typography/styles.ts'
    metadata:
      description: "Para garantir consistência tipográfica e centralizar a lógica de estilos de fonte, utilize a prop 'variant' no componente Typography em vez de definir 'font-size' manualmente."
      tags: ['styles', 'typography', 'variant', 'font-size']

  - id: ensure-text-contrast-ratio
    languages: [javascript, typescript]
    message: 'Garanta o contraste adequado para texto usando a propriedade contrast=true para fundos escuros ou use cores com uma razão de contraste de pelo menos 4.5:1.'
    severity: WARNING
    patterns:
      - pattern-either:
          - pattern-regex: "<Typography[^>]*(?!.*contrast=\\{true\\}).*>.*<\\/Typography>"
          - pattern-regex: "<Text[^>]*(?!.*contrast=\\{true\\}).*>.*<\\/Text>"
    paths:
      exclude:
        - '*.json'
        - '*.spec.*'
        - '*.test.*'
    metadata:
      description: 'Para garantir acessibilidade adequada, textos devem ter um contraste suficiente com o fundo. Use a propriedade contrast para elementos de texto em fundos escuros ou coloridos, ou certifique-se de que a razão de contraste atende às diretrizes WCAG.'
      tags: ['acessibilidade', 'contraste', 'texto', 'legibilidade']

  - id: ensure-proper-font-size-for-accessibility
    languages: [javascript, typescript]
    message: 'Evite tamanhos de fonte muito pequenos. Prefira usar variant="md" ou maior para texto principal.'
    severity: WARNING
    patterns:
      - pattern-regex: "variant=\"xs\""
    paths:
      exclude:
        - '*.json'
        - '*.spec.*'
        - '*.test.*'
        - '**/Caption/**'
    metadata:
      description: 'Fontes muito pequenas podem comprometer a legibilidade para usuários com deficiência visual. Use tamanhos de fonte adequados (preferência para md ou maior) para conteúdo principal.'
      tags: ['acessibilidade', 'tipografia', 'legibilidade', 'tamanho de fonte']
