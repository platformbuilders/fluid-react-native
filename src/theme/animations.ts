import { Animations } from '@platformbuilders/theme-toolkit';
import {
  LoadDotsAnimation,
  LoadDotsContrastAnimation,
  LoadingCircleAnimation,
  LoadingCircleContrastAnimation,
  LoadingProgressAnimation,
} from '../assets/animations';

export default {
  circularLoading: LoadingCircleAnimation,
  contrastCircularLoading: LoadingCircleContrastAnimation,
  buttonLoading: LoadDotsAnimation,
  contrastButtonLoading: LoadDotsContrastAnimation,
  linearLoading: LoadingProgressAnimation,
  contrastLinearLoading: LoadingProgressAnimation,
} as Animations;
