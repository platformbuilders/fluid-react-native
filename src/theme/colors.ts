import { darken, lighten } from 'polished';
import { Colors } from '../types';

const primary = '#C6B09E';
const secondary = '#805281';
const tertiary = '#2A2E39';
const accent = '#547E6A';
const disabled = '#eeeeee';

const colors: Colors = {
  text: '#212121',
  info: '#4096D1',
  success: '#357a38',
  warning: '#F5B800',
  failure: '#cc0000',
  disabled: {
    light: lighten(0.05, disabled),
    main: disabled,
    dark: darken(0.3, disabled),
    contrast: '#cccccc',
  },
  primary: {
    light: lighten(0.05, primary),
    main: primary,
    dark: darken(0.12, primary),
    contrast: '#ffffff',
  },
  secondary: {
    light: lighten(0.05, secondary),
    main: secondary,
    dark: darken(0.1, secondary),
    contrast: '#ffffff',
  },
  tertiary: {
    light: lighten(0.15, tertiary),
    main: tertiary,
    dark: darken(0.1, tertiary),
    contrast: '#ffffff',
  },
  accent: {
    light: lighten(0.05, accent),
    main: accent,
    dark: darken(0.05, accent),
    contrast: '#ffffff',
  },
};

export default colors;
