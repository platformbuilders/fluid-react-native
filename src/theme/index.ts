export { default as colors } from './colors';
export { default as spacing } from './spacing';
export { default as radius } from './radius';
export { default as animations } from './animations';
export { default as typography } from './typography';
