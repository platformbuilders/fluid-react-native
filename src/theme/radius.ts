import { moderateScale } from 'react-native-size-matters';
import { Radius } from '../types';

export default {
  smallRadius: `${moderateScale(3)}px`,
  mediumRadius: `${moderateScale(8)}px`,
  largeRadius: `${moderateScale(15)}px`,
  modalRadius: `${moderateScale(20)}px`,
  buttonRadius: `${moderateScale(7)}px`,
} as Radius;
