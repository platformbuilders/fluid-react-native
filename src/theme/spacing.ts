import { moderateScale } from 'react-native-size-matters';
import { isIphoneX, getStatusBarHeight } from 'react-native-iphone-x-helper';
import { Spacings } from '../types';

export default {
  topSpacing: `${getStatusBarHeight()}px`,
  bottomSpacing: `${isIphoneX() ? moderateScale(10) : 0}px`,
  sceneSpacing: `${moderateScale(24)}px`,
  sectionSpacing: `${moderateScale(40)}px`,
  minimumSpacing: `${moderateScale(4)}px`,
  smallSpacing: `${moderateScale(12)}px`,
  mediumSpacing: `${moderateScale(20)}px`,
  largeSpacing: `${moderateScale(24)}px`,
  giantSpacing: `${moderateScale(72)}px`,
} as Spacings;
