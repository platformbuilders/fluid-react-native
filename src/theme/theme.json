{
  "spacing": {
    "min": {
      "value": 1,
      "type": "spacing"
    },
    "xxs": {
      "value": 2,
      "type": "spacing"
    },
    "xs": {
      "value": 4,
      "type": "spacing"
    },
    "sm": {
      "value": 8,
      "type": "spacing"
    },
    "md": {
      "value": 16,
      "type": "spacing"
    },
    "lg": {
      "value": 24,
      "type": "spacing"
    },
    "xl": {
      "value": 32,
      "type": "spacing"
    },
    "xxl": {
      "value": 48,
      "type": "spacing"
    },
    "max": {
      "value": 64,
      "type": "spacing"
    }
  },
  "sizing": {
    "min": {
      "value": 1,
      "type": "sizing"
    },
    "xxs": {
      "value": 2,
      "type": "sizing"
    },
    "xs": {
      "value": 4,
      "type": "sizing"
    },
    "sm": {
      "value": 8,
      "type": "sizing"
    },
    "md": {
      "value": 16,
      "type": "sizing"
    },
    "lg": {
      "value": 24,
      "type": "sizing"
    },
    "xl": {
      "value": 32,
      "type": "sizing"
    },
    "xxl": {
      "value": 48,
      "type": "sizing"
    },
    "max": {
      "value": 64,
      "type": "sizing"
    }
  },
  "borderRadius": {
    "min": {
      "value": 0,
      "type": "borderRadius"
    },
    "xxs": {
      "value": 1,
      "type": "borderRadius"
    },
    "xs": {
      "value": 2,
      "type": "borderRadius"
    },
    "sm": {
      "value": 4,
      "type": "borderRadius"
    },
    "md": {
      "value": 6,
      "type": "borderRadius"
    },
    "lg": {
      "value": 8,
      "type": "borderRadius"
    },
    "xl": {
      "value": 16,
      "type": "borderRadius"
    },
    "xxl": {
      "value": 32,
      "type": "borderRadius"
    },
    "max": {
      "value": 999,
      "type": "borderRadius"
    }
  },
  "borderWidth": {
    "min": {
      "value": 0.5,
      "type": "borderWidth"
    },
    "xxs": {
      "value": 1,
      "type": "borderWidth"
    },
    "xs": {
      "value": 1.5,
      "type": "borderWidth"
    },
    "sm": {
      "value": 2,
      "type": "borderWidth"
    },
    "md": {
      "value": 3,
      "type": "borderWidth"
    },
    "lg": {
      "value": 4,
      "type": "borderWidth"
    },
    "xl": {
      "value": 6,
      "type": "borderWidth"
    },
    "xxl": {
      "value": 8,
      "type": "borderWidth"
    },
    "max": {
      "value": 12,
      "type": "borderWidth"
    }
  },
  "boxShadow": {
    "z0": {
      "value": {
        "x": 0,
        "y": 0,
        "spread": 0,
        "color": "",
        "blur": 0
      },
      "type": "boxShadow"
    },
    "z1": {
      "value": {
        "x": 0,
        "y": 1,
        "spread": 0,
        "color": "rgba(0, 0, 0, 0.1)",
        "blur": 3
      },
      "type": "boxShadow"
    },
    "z2": {
      "value": {
        "x": 0,
        "y": 4,
        "spread": -1,
        "color": "rgba(0, 0, 0, 0.1)",
        "blur": 6
      },
      "type": "boxShadow"
    },
    "z3": {
      "value": {
        "x": 0,
        "y": 10,
        "spread": -3,
        "color": "rgba(0, 0, 0, 0.1)",
        "blur": 15
      },
      "type": "boxShadow"
    },
    "z4": {
      "value": {
        "x": 0,
        "y": 20,
        "spread": -5,
        "color": "rgba(0, 0, 0, 0.1)",
        "blur": 25
      },
      "type": "boxShadow"
    },
    "z5": {
      "value": {
        "x": 0,
        "y": 25,
        "spread": -12,
        "color": "rgba(0, 0, 0, 0.2)",
        "blur": 50
      },
      "type": "boxShadow"
    }
  },
  "brand": {
    "primary": {
      "light": {
        "value": "#7684F9",
        "type": "color"
      },
      "main": {
        "value": "#4255EF",
        "type": "color"
      },
      "dark": {
        "value": "#1B288E",
        "type": "color"
      },
      "contrast": {
        "value": "#ffffff",
        "type": "color"
      }
    },
    "secondary": {
      "light": {
        "value": "#8BD6F8",
        "type": "color"
      },
      "main": {
        "value": "#42ADEF",
        "type": "color"
      },
      "dark": {
        "value": "#337BB7",
        "type": "color"
      },
      "contrast": {
        "value": "#ffffff",
        "type": "color"
      }
    },
    "tertiary": {
      "light": {
        "value": "#BA96F5",
        "type": "color"
      },
      "main": {
        "value": "#8442EF",
        "type": "color"
      },
      "dark": {
        "value": "#4A00DC",
        "type": "color"
      },
      "contrast": {
        "value": "#ffffff",
        "type": "color"
      }
    },
    "accent": {
      "light": {
        "value": "#FBD268",
        "type": "color"
      },
      "main": {
        "value": "#F49D19",
        "type": "color"
      },
      "dark": {
        "value": "#A1530A",
        "type": "color"
      },
      "contrast": {
        "value": "#351B02",
        "type": "color"
      }
    }
  },
  "success": {
    "light": {
      "value": "#F4FCD1",
      "type": "color"
    },
    "main": {
      "value": "#96C920",
      "type": "color"
    },
    "dark": {
      "value": "#396006",
      "type": "color"
    },
    "contrast": {
      "value": "#ffffff",
      "type": "color"
    }
  },
  "danger": {
    "light": {
      "value": "#FFE8D8",
      "type": "color",
      "description": "the 100 shade of a color"
    },
    "main": {
      "value": "#FF4F3F",
      "type": "color"
    },
    "dark": {
      "value": "#7A0C27",
      "type": "color",
      "description": "the 900s shade of a color"
    },
    "contrast": {
      "value": "#ffffff",
      "type": "color"
    }
  },
  "warning": {
    "light": {
      "value": "#FFF8CE",
      "type": "color"
    },
    "main": {
      "value": "#FFCA0C",
      "type": "color"
    },
    "dark": {
      "value": "#7A5502",
      "type": "color"
    },
    "contrast": {
      "value": "#261B00",
      "type": "color"
    }
  },
  "info": {
    "light": {
      "value": "#C2ECEC",
      "type": "color"
    },
    "main": {
      "value": "#42CCD4",
      "type": "color"
    },
    "dark": {
      "value": "#186F82",
      "type": "color"
    },
    "contrast": {
      "value": "#ffffff",
      "type": "color"
    }
  },
  "text": {
    "main": {
      "value": "#444444",
      "type": "color"
    },
    "dark": {
      "value": "#000",
      "type": "color"
    },
    "light": {
      "value": "#999999",
      "type": "color"
    },
    "contrast": {
      "value": "#fff",
      "type": "color"
    }
  },
  "background": {
    "z0": {
      "value": "#ffffff",
      "type": "color"
    },
    "z1": {
      "value": "#fafafa",
      "type": "color"
    },
    "z2": {
      "value": "#f0f0f0",
      "type": "color"
    },
    "z3": {
      "value": "#dddddd",
      "type": "color"
    },
    "z4": {
      "value": "#d1d1d1",
      "type": "color"
    }
  },
  "opacity": {
    "disabled": {
      "value": "30%",
      "type": "opacity"
    },
    "empty": {
      "value": "50%",
      "type": "opacity"
    },
    "overlay": {
      "value": "80%",
      "type": "opacity"
    }
  },
  "fontFamilies": {
    "heading": {
      "value": "Roboto",
      "type": "fontFamilies"
    },
    "body": {
      "value": "Roboto",
      "type": "fontFamilies"
    }
  },
  "lineHeights": {
    "heading": {
      "value": "110%",
      "type": "lineHeights"
    },
    "body": {
      "value": "140%",
      "type": "lineHeights"
    }
  },
  "letterSpacing": {
    "default": {
      "value": 0,
      "type": "letterSpacing"
    },
    "bigger": {
      "value": "105%",
      "type": "letterSpacing"
    }
  },
  "paragraphSpacing": {
    "h1": {
      "value": 32,
      "type": "paragraphSpacing"
    },
    "h2": {
      "value": 26,
      "type": "paragraphSpacing"
    }
  },
  "fontWeights": {
    "headingRegular": {
      "value": "Regular",
      "type": "fontWeights"
    },
    "headingBold": {
      "value": "Bold",
      "type": "fontWeights"
    },
    "bodyRegular": {
      "value": "Regular",
      "type": "fontWeights"
    },
    "bodyBold": {
      "value": "Bold",
      "type": "fontWeights"
    }
  },
  "fontSizes": {
    "min": {
      "value": 10,
      "type": "fontSizes"
    },
    "xxs": {
      "value": 11,
      "type": "fontSizes"
    },
    "xs": {
      "value": 13,
      "type": "fontSizes"
    },
    "sm": {
      "value": 14,
      "type": "fontSizes"
    },
    "md": {
      "value": 15,
      "type": "fontSizes"
    },
    "lg": {
      "value": 17,
      "type": "fontSizes"
    },
    "xl": {
      "value": 20,
      "type": "fontSizes"
    },
    "xxl": {
      "value": 24,
      "type": "fontSizes"
    },
    "max": {
      "value": 32,
      "type": "fontSizes"
    }
  },
  "themeRadius": {
    "button": {
      "value": 3,
      "type": "borderRadius"
    },
    "card": {
      "value": 6,
      "type": "borderRadius"
    },
    "input": {
      "value": 3,
      "type": "borderRadius"
    }
  },
  "typography": {
    "H1": {
      "Bold": {
        "type": "typography",
        "value": {
          "fontFamily": "$fontFamilies.heading",
          "fontWeight": "$fontWeights.headingBold",
          "lineHeight": "$lineHeights.heading",
          "fontSize": "$fontSizes.h1",
          "paragraphSpacing": "$paragraphSpacing.h1",
          "letterSpacing": "$letterSpacing.decreased"
        }
      },
      "Regular": {
        "type": "typography",
        "value": {
          "fontFamily": "$fontFamilies.heading",
          "fontWeight": "$fontWeights.headingRegular",
          "lineHeight": "$lineHeights.heading",
          "fontSize": "$fontSizes.h1",
          "paragraphSpacing": "$paragraphSpacing.h1",
          "letterSpacing": "$letterSpacing.decreased"
        }
      }
    },
    "H2": {
      "Bold": {
        "type": "typography",
        "value": {
          "fontFamily": "$fontFamilies.heading",
          "fontWeight": "$fontWeights.headingBold",
          "lineHeight": "$lineHeights.heading",
          "fontSize": "$fontSizes.h2",
          "paragraphSpacing": "$paragraphSpacing.h2",
          "letterSpacing": "$letterSpacing.decreased"
        }
      },
      "Regular": {
        "type": "typography",
        "value": {
          "fontFamily": "$fontFamilies.heading",
          "fontWeight": "$fontWeights.headingRegular",
          "lineHeight": "$lineHeights.heading",
          "fontSize": "$fontSizes.h2",
          "paragraphSpacing": "$paragraphSpacing.h2",
          "letterSpacing": "$letterSpacing.decreased"
        }
      }
    },
    "Body": {
      "type": "typography",
      "value": {
        "fontFamily": "$fontFamilies.body",
        "fontWeight": "$fontWeights.bodyRegular",
        "lineHeight": "$lineHeights.heading",
        "fontSize": "$fontSizes.body",
        "paragraphSpacing": "$paragraphSpacing.h2"
      }
    }
  }
}
