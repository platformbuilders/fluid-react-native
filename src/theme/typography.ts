import { TypographyTheme } from '../types/Typography';

export default {
  largeTitle: {
    fontSize: 34,
    lineHeight: 41,
  },
  title1: {
    fontSize: 28,
    lineHeight: 34,
  },
  title2: {
    fontSize: 22,
    lineHeight: 28,
  },
  title3: {
    fontSize: 20,
    lineHeight: 25,
  },
  title4: {
    fontSize: 18,
    lineHeight: 22,
  },
  headline: {
    fontSize: 17,
    lineHeight: 22,
  },
  body: {
    fontSize: 16,
    lineHeight: 22,
  },
  callout: {
    fontSize: 16,
    lineHeight: 21,
  },
  subhead: {
    fontSize: 15,
    lineHeight: 20,
  },
  footnote: {
    fontSize: 13,
    lineHeight: 18,
  },
  caption1: {
    fontSize: 12,
    lineHeight: 16,
  },
  caption2: {
    fontSize: 11,
    lineHeight: 13,
  },
} as TypographyTheme;
