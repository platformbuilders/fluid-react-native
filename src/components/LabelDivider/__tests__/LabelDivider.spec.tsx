import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components/native';
import Label from '..';

import theme from '../../../theme';

describe('<Label />', () => {
  it('should render Label', () => {
    const wrapper = renderer.create(
      <ThemeProvider theme={theme}>
        <Label text="label" />
      </ThemeProvider>,
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
