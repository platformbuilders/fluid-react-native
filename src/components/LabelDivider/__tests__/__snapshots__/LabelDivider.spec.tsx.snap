// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Label /> should render Label 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "marginBottom": 0,
        "marginTop": 0,
      },
    ]
  }
>
  <View
    style={
      [
        {
          "backgroundColor": "#99999990",
          "height": 1,
          "width": "33%",
        },
      ]
    }
  />
  <Text
    accessibilityLabel="label"
    lineHeightVariant="min"
    style={
      [
        {
          "color": "#444444",
          "fontSize": 15,
          "lineHeight": 18,
        },
        {
          "color": "#99999990",
          "fontWeight": "bold",
          "paddingBottom": 4,
          "paddingLeft": 4,
          "paddingRight": 4,
          "paddingTop": 4,
        },
      ]
    }
    variant="md"
  >
    label
  </Text>
  <View
    style={
      [
        {
          "backgroundColor": "#99999990",
          "height": 1,
          "width": "33%",
        },
      ]
    }
  />
</View>
`;
